app/controllers/users/settings/user_teams_controller.rb

```diff
@@ -150,7 +150,7 @@ def load_user_team
         # Don't allow the user to modify UserTeam-s if he's not admin,
         # unless he/she is modifying his/her UserTeam
         if current_user != @user_t.user &&
-           !is_admin_of_team(@user_t.team)
+           !can_read_team?(@user_t.team)
```

Comment:
Here the same, I would make separate methods `check_leave_team_permission` and `check_manage_user_team_permission`.

Reply:
And here use `can_manage_team_users?`